File: agent/src/main/java/io/github/vicitori/threading/highlighter/agent/common/TraceRecord.java
package io.github.vicitori.threading.highlighter.agent.common;

/**
 * A single stack-trace frame captured by the agent.
 * JSON field names must match the plugin's deserialization contract.
 */
public record TraceRecord(
        String className,
        String methodName,
        String fileName,
        int lineNumber,
        long lastSeenTimestampEpochMillis
) {
    public String getKey() {
        return className + "#" + methodName + "@" + lineNumber;
    }
}

File: common/src/main/kotlin/io/github/vicitori/threading/highlighter/common/trace/TraceRecord.kt
package io.github.vicitori.threading.highlighter.common.trace

import kotlinx.serialization.Serializable

@Serializable
data class TraceRecord(
    val className: String,
    val methodName: String,
    val fileName: String?,
    val lineNumber: Int,
    val lastSeenTimestampEpochMillis: Long
) {
    @JvmName("getKey")
    fun getKey(): String = "$className#$methodName@$lineNumber"
}
